### PR TITLE
Lots of mb_string functions to the rescue!

### PR DESCRIPTION
**Pull request by Camwyn**
🎫 https://central.tri.be/issues/70379

Convert multibyte characters to uppercase without choking on them.
Switch to using the `post_title` instead of the `post_name`, as post_name is urlencoded

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space after the first parenthesis | [Post_Root_Pool.php:91](https://github.com/moderntribe/tribe-common/blob/2f00f0d35e23e7a4f01bf64bc168b9fa69a57e81/src/Tribe/Utils/Post_Root_Pool.php#L91)
:x: | There must be a single space before the last parenthesis | [Post_Root_Pool.php:91](https://github.com/moderntribe/tribe-common/blob/2f00f0d35e23e7a4f01bf64bc168b9fa69a57e81/src/Tribe/Utils/Post_Root_Pool.php#L91)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Utils/Post_Root_Pool.php [hunk 1]
```diff
@@ -88,15 +91,32 @@ public function get_pool_transient_name() {
...
+		}
+
+		return $letter;
+	}
+
+	/**
+	 * @param $string
+	 *
+	 * @return string
+	 * @deprecated
```
Comment (Camwyn) [inline src/Tribe/Utils/Post_Root_Pool.php]: This is handled in the new functions.

File: src/functions/multibyte_helpers.php [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		// If it's a number, don't try to convert it
+		if ( is_numeric( $string ) ) {
+			return $string;
+		}
+
+		if ( function_exists( 'mb_strtolower' ) ) {
+			$encoding = tribe_detect_encoding( $string );
+			// we test for encoding and pass it if we get it
+			if ( $encoding ) {
+				return mb_strtolower(  $string, $encoding );
```
Comment (bordoni) [inline src/functions/multibyte_helpers.php]: Double spacing here

File: src/functions/multibyte_helpers.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+			}
+
+			return mb_strtolower(  $string );
```
Comment (bordoni) [inline src/functions/multibyte_helpers.php]: Double spacing here

File: src/functions/multibyte_helpers.php [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+		// If it's a number, don't try to convert it
+		if ( is_numeric( $string ) ) {
+			return $string;
+		}
+
+		if ( function_exists( 'mb_strtoupper' ) ) {
+			$encoding = tribe_detect_encoding( $string );
+			// we test for encoding and pass it if we get it
+			if ( $encoding ) {
+				return mb_strtoupper(  $string, $encoding );
```
Comment (bordoni) [inline src/functions/multibyte_helpers.php]: Double spacing here

**Comment by Camwyn** [comment]
Whitespace fixed!

**Comment by lucatume** [comment]
From the changes to the `autoload_` files it seems like `composer install --no-dev` was not called before the commit.
Do `rm -rf vendor && git checkout vendor && composer install --no-dev`

File: vendor/composer/autoload_static.php [hunk 5]
```diff
@@ -26,13 +26,6 @@ class ComposerStaticInit453516e196e1be4c43fa495b9bb9d572
     public static $classMap = array (
         'WP_Async_Request' => __DIR__ . '/..' . '/a5hleyrich/wp-background-processing/classes/wp-async-request.php',
         'WP_Background_Process' => __DIR__ . '/..' . '/a5hleyrich/wp-background-processing/classes/wp-background-process.php',
-        'tad_DI52_Container' => __DIR__ . '/..' . '/lucatume/di52/src/tad/DI52/Container.php',
```
Comment (lucatume) [inline vendor/composer/autoload_static.php]: This should not be removed...

Comment (Camwyn) [reply vendor/composer/autoload_static.php]: hm...

Comment (Camwyn) [reply vendor/composer/autoload_static.php]: No matter how I try to work composer, it keeps getting removed :(

**Comment by bordoni** [comment]
Closing this so far, because of #731

**Pull request closed without merging**